src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -80,9 +80,10 @@ internal ConversionsBase WithNullability(bool includeNullability)
         /// Determines if the source expression is convertible to the destination type via
         /// any built-in or user-defined implicit conversion.
         /// </summary>
-        public Conversion ClassifyImplicitConversionFromExpression(BoundExpression sourceExpression, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        public Conversion ClassifyImplicitConversionFromExpression(BoundExpression sourceExpression, CSharpCompilation compilation, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
```

Comment:
> Was it considered to make 'compilation' a field on \`ConversionsBase\` instead of adding a parameter?

```ConversionBase``` intentionally doesn't depend on any binding context. Classification of conversions between types doesn't depend on it. I tried to move all classification of conversion from expression to ```Conversions``` (it has Binder), unfortunately implementation of many methods is structured in such a way that it bundles both classifications (from type and from expression) together and conditionally looks at expression when it is passed in. That effort quickly became rather complicated because one of the goals was to avoid code duplication and minimize code changes. Then I tried the approach where compilation is passed along with expression and that became much easier to implement and I think it is much easier to review.

Reply:
Makes sense. Thanks for the information.